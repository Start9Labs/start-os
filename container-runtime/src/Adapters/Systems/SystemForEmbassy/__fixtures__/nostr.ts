export default {
  "tor-address": {
    name: "Tor Address",
    description: "The Tor address of the network interface",
    type: "pointer",
    subtype: "package",
    "package-id": "nostr-wallet-connect",
    target: "tor-address",
    interface: "main",
  },
  "lan-address": {
    name: "LAN Address",
    description: "The LAN address of the network interface",
    type: "pointer",
    subtype: "package",
    "package-id": "nostr-wallet-connect",
    target: "lan-address",
    interface: "main",
  },
  "nostr-relay": {
    type: "string",
    name: "Nostr Relay",
    default: "wss://relay.getalby.com/v1",
    description: "The Nostr Relay to use for Nostr Wallet Connect connections",
    copyable: true,
    nullable: false,
  },
}
