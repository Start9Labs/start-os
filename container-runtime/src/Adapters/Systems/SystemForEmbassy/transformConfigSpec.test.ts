import {
  matchOldConfigSpec,
  matchOldValueSpecList,
  transformConfigSpec,
} from "./transformConfigSpec"
import fixtureEmbassyPagesConfig from "./__fixtures__/embassyPagesConfig"
import fixtureRTLConfig from "./__fixtures__/rtlConfig"
import searNXG from "./__fixtures__/searNXG"
import bitcoind from "./__fixtures__/bitcoind"
import nostr from "./__fixtures__/nostr"
import nostrConfig2 from "./__fixtures__/nostrConfig2"

describe("transformConfigSpec", () => {
  test("matchOldConfigSpec(embassyPages.homepage.variants[web-page])", () => {
    matchOldConfigSpec.unsafeCast(
      fixtureEmbassyPagesConfig.homepage.variants["web-page"],
    )
  })
  test("matchOldConfigSpec(embassyPages)", () => {
    matchOldConfigSpec.unsafeCast(fixtureEmbassyPagesConfig)
  })
  test("transformConfigSpec(embassyPages)", () => {
    const spec = matchOldConfigSpec.unsafeCast(fixtureEmbassyPagesConfig)
    expect(transformConfigSpec(spec)).toMatchSnapshot()
  })

  test("matchOldConfigSpec(RTL.nodes)", () => {
    matchOldValueSpecList.unsafeCast(fixtureRTLConfig.nodes)
  })
  test("matchOldConfigSpec(RTL)", () => {
    matchOldConfigSpec.unsafeCast(fixtureRTLConfig)
  })
  test("transformConfigSpec(RTL)", () => {
    const spec = matchOldConfigSpec.unsafeCast(fixtureRTLConfig)
    expect(transformConfigSpec(spec)).toMatchSnapshot()
  })

  test("transformConfigSpec(searNXG)", () => {
    const spec = matchOldConfigSpec.unsafeCast(searNXG)
    expect(transformConfigSpec(spec)).toMatchSnapshot()
  })
  test("transformConfigSpec(bitcoind)", () => {
    const spec = matchOldConfigSpec.unsafeCast(bitcoind)
    expect(transformConfigSpec(spec)).toMatchSnapshot()
  })
  test("transformConfigSpec(nostr)", () => {
    const spec = matchOldConfigSpec.unsafeCast(nostr)
    expect(transformConfigSpec(spec)).toMatchSnapshot()
  })
  test("transformConfigSpec(nostr2)", () => {
    const spec = matchOldConfigSpec.unsafeCast(nostrConfig2)
    expect(transformConfigSpec(spec)).toMatchSnapshot()
  })
})
