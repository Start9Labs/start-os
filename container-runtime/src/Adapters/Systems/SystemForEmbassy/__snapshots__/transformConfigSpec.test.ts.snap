// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformConfigSpec transformConfigSpec(RTL) 1`] = `
{
  "password": {
    "default": {
      "charset": "a-z,A-Z,0-9",
      "len": 22,
    },
    "description": "The password for your Ride the Lightning dashboard",
    "disabled": false,
    "generate": null,
    "immutable": false,
    "inputmode": "text",
    "masked": true,
    "maxLength": null,
    "minLength": null,
    "name": "Password",
    "patterns": [],
    "placeholder": null,
    "required": true,
    "type": "text",
    "warning": null,
  },
}
`;

exports[`transformConfigSpec transformConfigSpec(bitcoind) 1`] = `
{
  "advanced": {
    "description": "Advanced Settings",
    "name": "Advanced",
    "spec": {
      "blockfilters": {
        "description": "Settings for storing and serving compact block filters",
        "name": "Block Filters",
        "spec": {
          "blockfilterindex": {
            "default": true,
            "description": "Generate Compact Block Filters during initial sync (IBD) to enable 'getblockfilter' RPC. This is useful if dependent services need block filters to efficiently scan for addresses/transactions etc.",
            "disabled": false,
            "immutable": false,
            "name": "Compute Compact Block Filters (BIP158)",
            "type": "toggle",
            "warning": null,
          },
          "peerblockfilters": {
            "default": false,
            "description": "Serve Compact Block Filters as a peer service to other nodes on the network. This is useful if you wish to connect an SPV client to your node to make it efficient to scan transactions without having to download all block data.  'Compute Compact Block Filters (BIP158)' is required.",
            "disabled": false,
            "immutable": false,
            "name": "Serve Compact Block Filters to Peers (BIP157)",
            "type": "toggle",
            "warning": null,
          },
        },
        "type": "object",
        "warning": null,
      },
      "bloomfilters": {
        "description": "Setting for serving Bloom Filters",
        "name": "Bloom Filters (BIP37)",
        "spec": {
          "peerbloomfilters": {
            "default": false,
            "description": "Peers have the option of setting filters on each connection they make after the version handshake has completed. Bloom filters are for clients implementing SPV (Simplified Payment Verification) that want to check that block headers  connect together correctly, without needing to verify the full blockchain.  The client must trust that the transactions in the chain are in fact valid.  It is highly recommended AGAINST using for anything except Bisq integration.",
            "disabled": false,
            "immutable": false,
            "name": "Serve Bloom Filters to Peers",
            "type": "toggle",
            "warning": "This is ONLY for use with Bisq integration, please use Block Filters for all other applications.",
          },
        },
        "type": "object",
        "warning": null,
      },
      "dbcache": {
        "default": null,
        "description": "How much RAM to allocate for caching the TXO set. Higher values improve syncing performance, but increase your chance of using up all your system's memory or corrupting your database in the event of an ungraceful shutdown. Set this high but comfortably below your system's total RAM during IBD, then turn down to 450 (or leave blank) once the sync completes.",
        "disabled": false,
        "immutable": false,
        "integer": true,
        "max": null,
        "min": null,
        "name": "Database Cache",
        "placeholder": null,
        "required": false,
        "step": null,
        "type": "number",
        "units": "MiB",
        "warning": "WARNING: Increasing this value results in a higher chance of ungraceful shutdowns, which can leave your node unusable if it happens during the initial block download. Use this setting with caution. Be sure to set this back to the default (450 or leave blank) once your node is synced. DO NOT press the STOP button if your dbcache is large. Instead, set this number back to the default, hit save, and wait for bitcoind to restart on its own.",
      },
      "mempool": {
        "description": "Mempool Settings",
        "name": "Mempool",
        "spec": {
          "datacarrier": {
            "default": true,
            "description": "Relay transactions with OP_RETURN outputs",
            "disabled": false,
            "immutable": false,
            "name": "Relay OP_RETURN Transactions",
            "type": "toggle",
            "warning": null,
          },
          "datacarriersize": {
            "default": 83,
            "description": "Maximum size of data in OP_RETURN outputs to relay",
            "disabled": false,
            "immutable": false,
            "integer": true,
            "max": 10000,
            "min": null,
            "name": "Max OP_RETURN Size",
            "placeholder": null,
            "required": true,
            "step": null,
            "type": "number",
            "units": "bytes",
            "warning": null,
          },
          "maxmempool": {
            "default": 300,
            "description": "Keep the transaction memory pool below <n> megabytes.",
            "disabled": false,
            "immutable": false,
            "integer": true,
            "max": null,
            "min": 1,
            "name": "Max Mempool Size",
            "placeholder": null,
            "required": true,
            "step": null,
            "type": "number",
            "units": "MiB",
            "warning": null,
          },
          "mempoolexpiry": {
            "default": 336,
            "description": "Do not keep transactions in the mempool longer than <n> hours.",
            "disabled": false,
            "immutable": false,
            "integer": true,
            "max": null,
            "min": 1,
            "name": "Mempool Expiration",
            "placeholder": null,
            "required": true,
            "step": null,
            "type": "number",
            "units": "Hr",
            "warning": null,
          },
          "mempoolfullrbf": {
            "default": true,
            "description": "Policy for your node to use for relaying and mining unconfirmed transactions.  For details, see https://github.com/bitcoin/bitcoin/blob/master/doc/release-notes/release-notes-24.0.1.md#notice-of-new-option-for-transaction-replacement-policies",
            "disabled": false,
            "immutable": false,
            "name": "Enable Full RBF",
            "type": "toggle",
            "warning": null,
          },
          "permitbaremultisig": {
            "default": true,
            "description": "Relay non-P2SH multisig transactions",
            "disabled": false,
            "immutable": false,
            "name": "Permit Bare Multisig",
            "type": "toggle",
            "warning": null,
          },
          "persistmempool": {
            "default": true,
            "description": "Save the mempool on shutdown and load on restart.",
            "disabled": false,
            "immutable": false,
            "name": "Persist Mempool",
            "type": "toggle",
            "warning": null,
          },
        },
        "type": "object",
        "warning": null,
      },
      "peers": {
        "description": "Peer Connection Settings",
        "name": "Peers",
        "spec": {
          "addnode": {
            "default": [],
            "description": "Add addresses of nodes to connect to.",
            "disabled": false,
            "maxLength": null,
            "minLength": null,
            "name": "Add Nodes",
            "spec": {
              "displayAs": null,
              "spec": {
                "hostname": {
                  "default": null,
                  "description": "Domain or IP address of bitcoin peer",
                  "disabled": false,
                  "generate": null,
                  "immutable": false,
                  "inputmode": "text",
                  "masked": false,
                  "maxLength": null,
                  "minLength": null,
                  "name": "Hostname",
                  "patterns": [
                    {
                      "description": "Must be either a domain name, or an IPv4 or IPv6 address. Do not include protocol scheme (eg 'http://') or port.",
                      "regex": "(^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$)|((^(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))$)|(^[a-z2-7]{16}\\.onion$)|(^([a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?\\.)+[a-z0-9][a-z0-9-]{0,61}[a-z0-9]$))",
                    },
                  ],
                  "placeholder": null,
                  "required": true,
                  "type": "text",
                  "warning": null,
                },
                "port": {
                  "default": null,
                  "description": "Port that peer is listening on for inbound p2p connections",
                  "disabled": false,
                  "immutable": false,
                  "integer": true,
                  "max": 65535,
                  "min": null,
                  "name": "Port",
                  "placeholder": null,
                  "required": false,
                  "step": null,
                  "type": "number",
                  "units": null,
                  "warning": null,
                },
              },
              "type": "object",
              "uniqueBy": null,
            },
            "type": "list",
            "warning": null,
          },
          "listen": {
            "default": true,
            "description": "Allow other nodes to find your server on the network.",
            "disabled": false,
            "immutable": false,
            "name": "Make Public",
            "type": "toggle",
            "warning": null,
          },
          "onlyconnect": {
            "default": false,
            "description": "Only connect to specified peers.",
            "disabled": false,
            "immutable": false,
            "name": "Disable Peer Discovery",
            "type": "toggle",
            "warning": null,
          },
          "onlyonion": {
            "default": false,
            "description": "Only connect to peers over Tor.",
            "disabled": false,
            "immutable": false,
            "name": "Disable Clearnet",
            "type": "toggle",
            "warning": null,
          },
          "v2transport": {
            "default": false,
            "description": "Enable or disable the use of BIP324 V2 P2P transport protocol.",
            "disabled": false,
            "immutable": false,
            "name": "Use V2 P2P Transport Protocol",
            "type": "toggle",
            "warning": null,
          },
        },
        "type": "object",
        "warning": null,
      },
      "pruning": {
        "default": "disabled",
        "description": "- Disabled: Disable pruning
- Automatic: Limit blockchain size on disk to a certain number of megabytes
",
        "disabled": false,
        "immutable": false,
        "name": "Pruning Mode",
        "type": "union",
        "variants": {
          "automatic": {
            "name": "Automatic",
            "spec": {
              "size": {
                "default": 550,
                "description": "Limit of blockchain size on disk.",
                "disabled": false,
                "immutable": false,
                "integer": true,
                "max": 999999,
                "min": 550,
                "name": "Max Chain Size",
                "placeholder": null,
                "required": true,
                "step": null,
                "type": "number",
                "units": "MiB",
                "warning": "Increasing this value will require re-syncing your node.",
              },
            },
          },
          "disabled": {
            "name": "Disabled",
            "spec": {},
          },
        },
        "warning": null,
      },
    },
    "type": "object",
    "warning": null,
  },
  "coinstatsindex": {
    "default": false,
    "description": "Enabling Coinstats Index reduces the time for the gettxoutsetinfo RPC to complete at the cost of using additional disk space",
    "disabled": false,
    "immutable": false,
    "name": "Coinstats Index",
    "type": "toggle",
    "warning": null,
  },
  "rpc": {
    "description": "RPC configuration options.",
    "name": "RPC Settings",
    "spec": {
      "advanced": {
        "description": "Advanced RPC Settings",
        "name": "Advanced",
        "spec": {
          "auth": {
            "default": [],
            "description": "Username and hashed password for JSON-RPC connections. RPC clients connect using the usual http basic authentication.",
            "disabled": false,
            "maxLength": null,
            "minLength": null,
            "name": "Authorization",
            "spec": {
              "generate": null,
              "inputmode": "text",
              "masked": false,
              "maxLength": null,
              "minLength": null,
              "patterns": [
                {
                  "description": "Each item must be of the form "<USERNAME>:<SALT>$<HASH>".",
                  "regex": "^[a-zA-Z0-9_-]+:([0-9a-fA-F]{2})+\\$([0-9a-fA-F]{2})+$",
                },
              ],
              "placeholder": null,
              "type": "text",
            },
            "type": "list",
            "warning": null,
          },
          "servertimeout": {
            "default": 30,
            "description": "Number of seconds after which an uncompleted RPC call will time out.",
            "disabled": false,
            "immutable": false,
            "integer": true,
            "max": 300,
            "min": 5,
            "name": "Rpc Server Timeout",
            "placeholder": null,
            "required": true,
            "step": null,
            "type": "number",
            "units": "seconds",
            "warning": null,
          },
          "threads": {
            "default": 16,
            "description": "Set the number of threads for handling RPC calls. You may wish to increase this if you are making lots of calls via an integration.",
            "disabled": false,
            "immutable": false,
            "integer": true,
            "max": 64,
            "min": 1,
            "name": "Threads",
            "placeholder": null,
            "required": true,
            "step": null,
            "type": "number",
            "units": null,
            "warning": null,
          },
          "workqueue": {
            "default": 128,
            "description": "Set the depth of the work queue to service RPC calls. Determines how long the backlog of RPC requests can get before it just rejects new ones.",
            "disabled": false,
            "immutable": false,
            "integer": true,
            "max": 256,
            "min": 8,
            "name": "Work Queue",
            "placeholder": null,
            "required": true,
            "step": null,
            "type": "number",
            "units": "requests",
            "warning": null,
          },
        },
        "type": "object",
        "warning": null,
      },
      "enable": {
        "default": true,
        "description": "Allow remote RPC requests.",
        "disabled": false,
        "immutable": false,
        "name": "Enable",
        "type": "toggle",
        "warning": null,
      },
      "password": {
        "default": {
          "charset": "a-z,2-7",
          "len": 20,
        },
        "description": "The password for connecting to Bitcoin over RPC.",
        "disabled": false,
        "generate": null,
        "immutable": false,
        "inputmode": "text",
        "masked": true,
        "maxLength": null,
        "minLength": null,
        "name": "RPC Password",
        "patterns": [
          {
            "description": "Must be alphanumeric (can contain underscore).",
            "regex": "^[a-zA-Z0-9_]+$",
          },
        ],
        "placeholder": null,
        "required": true,
        "type": "text",
        "warning": "You will need to restart all services that depend on Bitcoin.",
      },
      "username": {
        "default": "bitcoin",
        "description": "The username for connecting to Bitcoin over RPC.",
        "disabled": false,
        "generate": null,
        "immutable": false,
        "inputmode": "text",
        "masked": true,
        "maxLength": null,
        "minLength": null,
        "name": "Username",
        "patterns": [
          {
            "description": "Must be alphanumeric (can contain underscore).",
            "regex": "^[a-zA-Z0-9_]+$",
          },
        ],
        "placeholder": null,
        "required": true,
        "type": "text",
        "warning": "You will need to restart all services that depend on Bitcoin.",
      },
    },
    "type": "object",
    "warning": null,
  },
  "txindex": {
    "default": true,
    "description": "By enabling Transaction Index (txindex) Bitcoin Core will build a complete transaction index. This allows Bitcoin Core to access any transaction with commands like \`gettransaction\`.",
    "disabled": false,
    "immutable": false,
    "name": "Transaction Index",
    "type": "toggle",
    "warning": null,
  },
  "wallet": {
    "description": "Wallet Settings",
    "name": "Wallet",
    "spec": {
      "avoidpartialspends": {
        "default": true,
        "description": "Group outputs by address, selecting all or none, instead of selecting on a per-output basis. This improves privacy at the expense of higher transaction fees.",
        "disabled": false,
        "immutable": false,
        "name": "Avoid Partial Spends",
        "type": "toggle",
        "warning": null,
      },
      "discardfee": {
        "default": 0.0001,
        "description": "The fee rate (in BTC/kB) that indicates your tolerance for discarding change by adding it to the fee.",
        "disabled": false,
        "immutable": false,
        "integer": false,
        "max": 0.01,
        "min": null,
        "name": "Discard Change Tolerance",
        "placeholder": null,
        "required": true,
        "step": null,
        "type": "number",
        "units": "BTC/kB",
        "warning": null,
      },
      "enable": {
        "default": true,
        "description": "Load the wallet and enable wallet RPC calls.",
        "disabled": false,
        "immutable": false,
        "name": "Enable Wallet",
        "type": "toggle",
        "warning": null,
      },
    },
    "type": "object",
    "warning": null,
  },
  "zmq-enabled": {
    "default": true,
    "description": "The ZeroMQ interface is useful for some applications which might require data related to block and transaction events from Bitcoin Core. For example, LND requires ZeroMQ be enabled for LND to get the latest block data",
    "disabled": false,
    "immutable": false,
    "name": "ZeroMQ Enabled",
    "type": "toggle",
    "warning": null,
  },
}
`;

exports[`transformConfigSpec transformConfigSpec(embassyPages) 1`] = `
{
  "homepage": {
    "default": "welcome",
    "description": null,
    "disabled": false,
    "immutable": false,
    "name": "Type",
    "type": "union",
    "variants": {
      "index": {
        "name": "Table of Contents",
        "spec": {},
      },
      "redirect": {
        "name": "Redirect",
        "spec": {
          "target": {
            "default": null,
            "description": "The name of the subdomain to redirect users to. This must be a valid subdomain site within your Start9 Pages.",
            "disabled": false,
            "generate": null,
            "immutable": false,
            "inputmode": "text",
            "masked": false,
            "maxLength": null,
            "minLength": null,
            "name": "Target Subdomain",
            "patterns": [
              {
                "description": "May contain only lowercase characters and hyphens.",
                "regex": "^[a-z-]+$",
              },
            ],
            "placeholder": null,
            "required": true,
            "type": "text",
            "warning": null,
          },
        },
      },
      "web-page": {
        "name": "Web Page",
        "spec": {
          "folder": {
            "default": null,
            "description": "The path to the folder that contains the static files of your website. For example, a value of "projects/resume" would tell Start9 Pages to look for that folder path in the selected service.",
            "disabled": false,
            "generate": null,
            "immutable": false,
            "inputmode": "text",
            "masked": false,
            "maxLength": null,
            "minLength": null,
            "name": "Folder Path",
            "patterns": [
              {
                "description": "Must be a valid relative file path",
                "regex": "^(\\.|[a-zA-Z0-9_ -][a-zA-Z0-9_ .-]*|([a-zA-Z0-9_ .-][a-zA-Z0-9_ -]+\\.*)+)(/[a-zA-Z0-9_ -][a-zA-Z0-9_ .-]*|/([a-zA-Z0-9_ .-][a-zA-Z0-9_ -]+\\.*)+)*/?$",
              },
            ],
            "placeholder": "e.g. websites/resume",
            "required": true,
            "type": "text",
            "warning": null,
          },
          "source": {
            "default": "nextcloud",
            "description": "The service that contains your website files.",
            "disabled": false,
            "immutable": false,
            "name": "Folder Location",
            "type": "select",
            "values": {
              "filebrowser": "filebrowser",
              "nextcloud": "nextcloud",
            },
            "warning": null,
          },
        },
      },
      "welcome": {
        "name": "Welcome",
        "spec": {},
      },
    },
    "warning": null,
  },
  "subdomains": {
    "default": [],
    "description": "The websites you want to serve.",
    "disabled": false,
    "maxLength": null,
    "minLength": null,
    "name": "Subdomains",
    "spec": {
      "displayAs": "{{name}}",
      "spec": {
        "name": {
          "default": null,
          "description": "The subdomain of your Start9 Pages .onion address to host the website on. For example, a value of "me" would produce a website hosted at http://me.xxxxxx.onion.",
          "disabled": false,
          "generate": null,
          "immutable": false,
          "inputmode": "text",
          "masked": false,
          "maxLength": null,
          "minLength": null,
          "name": "Subdomain name",
          "patterns": [
            {
              "description": "May contain only lowercase characters and hyphens",
              "regex": "^[a-z-]+$",
            },
          ],
          "placeholder": null,
          "required": true,
          "type": "text",
          "warning": null,
        },
        "settings": {
          "default": "web-page",
          "description": null,
          "disabled": false,
          "immutable": false,
          "name": "Type",
          "type": "union",
          "variants": {
            "redirect": {
              "name": "Redirect",
              "spec": {
                "target": {
                  "default": null,
                  "description": "The subdomain of your Start9 Pages .onion address to redirect to. This should be the name of another subdomain on Start9 Pages. Leave empty to redirect to the homepage.",
                  "disabled": false,
                  "generate": null,
                  "immutable": false,
                  "inputmode": "text",
                  "masked": false,
                  "maxLength": null,
                  "minLength": null,
                  "name": "Target Subdomain",
                  "patterns": [
                    {
                      "description": "May contain only lowercase characters and hyphens.",
                      "regex": "^[a-z-]+$",
                    },
                  ],
                  "placeholder": null,
                  "required": true,
                  "type": "text",
                  "warning": null,
                },
              },
            },
            "web-page": {
              "name": "Web Page",
              "spec": {
                "folder": {
                  "default": null,
                  "description": "The path to the folder that contains the website files. For example, a value of "projects/resume" would tell Start9 Pages to look for that folder path in the selected service.",
                  "disabled": false,
                  "generate": null,
                  "immutable": false,
                  "inputmode": "text",
                  "masked": false,
                  "maxLength": null,
                  "minLength": null,
                  "name": "Folder Path",
                  "patterns": [
                    {
                      "description": "Must be a valid relative file path",
                      "regex": "^(\\.|[a-zA-Z0-9_ -][a-zA-Z0-9_ .-]*|([a-zA-Z0-9_ .-][a-zA-Z0-9_ -]+\\.*)+)(/[a-zA-Z0-9_ -][a-zA-Z0-9_ .-]*|/([a-zA-Z0-9_ .-][a-zA-Z0-9_ -]+\\.*)+)*/?$",
                    },
                  ],
                  "placeholder": "e.g. websites/resume",
                  "required": true,
                  "type": "text",
                  "warning": null,
                },
                "source": {
                  "default": "nextcloud",
                  "description": "The service that contains your website files.",
                  "disabled": false,
                  "immutable": false,
                  "name": "Folder Location",
                  "type": "select",
                  "values": {
                    "filebrowser": "filebrowser",
                    "nextcloud": "nextcloud",
                  },
                  "warning": null,
                },
              },
            },
          },
          "warning": null,
        },
      },
      "type": "object",
      "uniqueBy": "name",
    },
    "type": "list",
    "warning": null,
  },
}
`;

exports[`transformConfigSpec transformConfigSpec(nostr) 1`] = `
{
  "nostr-relay": {
    "default": "wss://relay.getalby.com/v1",
    "description": "The Nostr Relay to use for Nostr Wallet Connect connections",
    "disabled": false,
    "generate": null,
    "immutable": false,
    "inputmode": "text",
    "masked": false,
    "maxLength": null,
    "minLength": null,
    "name": "Nostr Relay",
    "patterns": [],
    "placeholder": null,
    "required": true,
    "type": "text",
    "warning": null,
  },
}
`;

exports[`transformConfigSpec transformConfigSpec(nostr2) 1`] = `
{
  "relay-type": {
    "default": "private",
    "description": "Private or public. A private relay (highly recommended) restricts write access to specific pubkeys. Anyone can write to a public relay.",
    "disabled": false,
    "immutable": false,
    "name": "Relay Type",
    "type": "union",
    "variants": {
      "private": {
        "name": "Private",
        "spec": {
          "pubkey_whitelist": {
            "default": [],
            "description": "A list of pubkeys that are permitted to publish through your relay. A minimum, you need to enter your own Nostr hex (not npub) pubkey. Go to https://damus.io/key/ to convert from npub to hex.",
            "disabled": false,
            "maxLength": null,
            "minLength": 1,
            "name": "Pubkey Whitelist (hex)",
            "spec": {
              "generate": null,
              "inputmode": "text",
              "masked": false,
              "maxLength": null,
              "minLength": null,
              "patterns": [
                {
                  "description": "Must be a valid 64-digit hexadecimal value (ie a Nostr hex pubkey, not an npub). Go to https://damus.io/key/ to convert npub to hex.",
                  "regex": "[0-9a-fA-F]{64}",
                },
              ],
              "placeholder": "hex (not npub) pubkey",
              "type": "text",
            },
            "type": "list",
            "warning": null,
          },
        },
      },
      "public": {
        "name": "Public",
        "spec": {
          "info": {
            "description": "General public info about your relay",
            "name": "Relay Info",
            "spec": {
              "contact": {
                "default": null,
                "description": "The email address of the relay administrator",
                "disabled": false,
                "generate": null,
                "immutable": false,
                "inputmode": "text",
                "masked": false,
                "maxLength": null,
                "minLength": null,
                "name": "Admin contact email",
                "patterns": [
                  {
                    "description": "Must be a valid email address.",
                    "regex": "[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+.[a-zA-Z0-9-.]+",
                  },
                ],
                "placeholder": null,
                "required": false,
                "type": "text",
                "warning": null,
              },
              "description": {
                "default": null,
                "description": "A more detailed description for your relay",
                "disabled": false,
                "generate": null,
                "immutable": false,
                "inputmode": "text",
                "masked": false,
                "maxLength": null,
                "minLength": null,
                "name": "Relay Description",
                "patterns": [
                  {
                    "description": "Must be at least 6 character and no more than 256 characters",
                    "regex": ".{6,256}",
                  },
                ],
                "placeholder": "The best relay in town",
                "required": false,
                "type": "text",
                "warning": null,
              },
              "name": {
                "default": null,
                "description": "Your relay's human-readable identifier",
                "disabled": false,
                "generate": null,
                "immutable": false,
                "inputmode": "text",
                "masked": false,
                "maxLength": null,
                "minLength": null,
                "name": "Relay Name",
                "patterns": [
                  {
                    "description": "Must be at least 3 character and no more than 32 characters",
                    "regex": ".{3,32}",
                  },
                ],
                "placeholder": "Bob's Public Relay",
                "required": false,
                "type": "text",
                "warning": null,
              },
              "pubkey": {
                "default": null,
                "description": "The Nostr hex (not npub) pubkey of the relay administrator",
                "disabled": false,
                "generate": null,
                "immutable": false,
                "inputmode": "text",
                "masked": false,
                "maxLength": null,
                "minLength": null,
                "name": "Admin contact pubkey (hex)",
                "patterns": [
                  {
                    "description": "Must be a valid 64-digit hexadecimal value (ie a Nostr hex pubkey, not an npub). Go to https://damus.io/key/ to convert npub to hex.",
                    "regex": "[0-9a-fA-F]{64}",
                  },
                ],
                "placeholder": "hex (not npub) pubkey",
                "required": false,
                "type": "text",
                "warning": null,
              },
            },
            "type": "object",
            "warning": null,
          },
          "limits": {
            "description": "Data limits to protect your relay from using too many resources",
            "name": "Limits",
            "spec": {
              "event_kind_blacklist": {
                "default": [],
                "description": "Events with these kinds will be discarded. For a list of event kinds, see here: https://github.com/nostr-protocol/nips#event-kinds",
                "disabled": false,
                "maxLength": null,
                "minLength": null,
                "name": "Event Kind Blacklist",
                "spec": {
                  "generate": null,
                  "inputmode": "text",
                  "masked": false,
                  "maxLength": null,
                  "minLength": null,
                  "patterns": [
                    {
                      "description": "Integral number type",
                      "regex": "[0-9]+",
                    },
                  ],
                  "placeholder": "30023",
                  "type": "text",
                },
                "type": "list",
                "warning": null,
              },
              "max_blocking_threads": {
                "default": 16,
                "description": "Maximum number of blocking threads used for database connections.",
                "disabled": false,
                "immutable": false,
                "integer": true,
                "max": null,
                "min": null,
                "name": "Max Blocking Threads",
                "placeholder": null,
                "required": true,
                "step": null,
                "type": "number",
                "units": "threads",
                "warning": null,
              },
              "max_event_bytes": {
                "default": 131072,
                "description": "Limit the maximum size of an EVENT message. Set to 0 for unlimited",
                "disabled": false,
                "immutable": false,
                "integer": true,
                "max": null,
                "min": null,
                "name": "Max Event Size",
                "placeholder": null,
                "required": true,
                "step": null,
                "type": "number",
                "units": "bytes",
                "warning": null,
              },
              "max_ws_frame_bytes": {
                "default": 131072,
                "description": "Maximum WebSocket frame size in bytes.",
                "disabled": false,
                "immutable": false,
                "integer": true,
                "max": null,
                "min": null,
                "name": "Max Websocket Frame Size",
                "placeholder": null,
                "required": true,
                "step": null,
                "type": "number",
                "units": "bytes",
                "warning": null,
              },
              "max_ws_message_bytes": {
                "default": 131072,
                "description": "Maximum WebSocket message in bytes.",
                "disabled": false,
                "immutable": false,
                "integer": true,
                "max": null,
                "min": null,
                "name": "Max Websocket Message Size",
                "placeholder": null,
                "required": true,
                "step": null,
                "type": "number",
                "units": "bytes",
                "warning": null,
              },
              "messages_per_sec": {
                "default": 2,
                "description": "Limit events created per second, averaged over one minute. Note: this is for the server as a whole, not per connection.",
                "disabled": false,
                "immutable": false,
                "integer": true,
                "max": null,
                "min": 1,
                "name": "Messages Per Second Limit",
                "placeholder": null,
                "required": true,
                "step": null,
                "type": "number",
                "units": "messages/sec",
                "warning": null,
              },
              "subscriptions_per_min": {
                "default": 10,
                "description": "Limit client subscriptions created per second, averaged over one minute. Strongly recommended to set this to a low value such as 10 to ensure fair service.",
                "disabled": false,
                "immutable": false,
                "integer": true,
                "max": null,
                "min": 1,
                "name": "Subscriptions Per Minute Limit",
                "placeholder": null,
                "required": true,
                "step": null,
                "type": "number",
                "units": "subscriptions",
                "warning": null,
              },
            },
            "type": "object",
            "warning": null,
          },
        },
      },
    },
    "warning": null,
  },
}
`;

exports[`transformConfigSpec transformConfigSpec(searNXG) 1`] = `
{
  "enable-metrics": {
    "default": true,
    "description": "Your SearXNG instance will collect anonymous stats about its own usage and performance. You can view these metrics by appending \`/stats\` or \`/stats/errors\` to your SearXNG URL.",
    "disabled": false,
    "immutable": false,
    "name": "Enable Stats",
    "type": "toggle",
    "warning": null,
  },
  "instance-name": {
    "default": "My SearXNG Engine",
    "description": "Enter a name for your SearXNG instance. This is the name that will be listed if you want to share your SearXNG engine publicly.",
    "disabled": false,
    "generate": null,
    "immutable": false,
    "inputmode": "text",
    "masked": false,
    "maxLength": null,
    "minLength": null,
    "name": "SearXNG Instance Name",
    "patterns": [],
    "placeholder": "Uncle Jim SearXNG Engine",
    "required": true,
    "type": "text",
    "warning": null,
  },
  "tor-url": {
    "default": false,
    "description": "Activates the utilization of a .onion address as the primary URL, particularly beneficial for publicly hosted instances over the Tor network.",
    "disabled": false,
    "immutable": false,
    "name": "Enable Tor address as the base URL",
    "type": "toggle",
    "warning": null,
  },
}
`;
